### ISSUE TITLE
XPath in an InsertBefore transform is using a trait of the parent node that's been transformed into something else

### ISSUE THREAD

**Issue by mhenr18** [CameronWills/FatAntelope#2]
(Side note: This tool is really cool! Out of 160 or so web.configs that I'm diffing from a common base and then applying a transform to as a test, only one of those files broke with this particular issue. Definitely will be a _massive_ time saver for us)

The exception is as follows:

```
An unhandled exception of type 'Microsoft.Web.XmlTransform.XmlNodeException' occurred in Microsoft.Web.XmlTransform.dll

Additional information: No element in the source document matches '/configuration/system.webServer/rewrite/rules/rule[(@name='Slideshow')]/*[1]'
```

From what I can see from my quick dig through the source and generated transform, the relevant snippet of the transform that's breaking is:

``` xml
<!-- ... -->
<rule xdt:Locator="Condition(2)" name="foo" xdt:Transform="SetAttributes(name)">
    <conditions logicalGrouping="MatchAll" xdt:Transform="InsertBefore(/configuration/system.webServer/rewrite/rules/rule[(@name='Slideshow')]/*[1])">
    <!-- ... -->
</rule>
<!-- ... -->
```

As you can see, it's transforming the name of the `<rule>` to `foo`, but then a nested transform has an XPath that's trying to refer to the `<rule>` by its old name of `Slideshow`.

I can reproduce this issue with the following files:

base.xml

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <rewrite>
            <rules>
                <rule name="Redirect to https" stopProcessing="true">
                    <match url="(.*)"/>
                    <conditions>
                        <add input="{HTTPS}" pattern="off" ignoreCase="true"/>
                    </conditions>
                    <action type="Redirect" url="https://{HTTP_HOST}{REQUEST_URI}" redirectType="Permanent"/>
                </rule>
                <rule name="Slideshow" stopProcessing="true">
                    <match url="^s/([^/]*)/([^/]*)(/)?(.*)?" ignoreCase="true"/>
                    <action type="Rewrite" url="slides/page.aspx?view={R:1}&amp;site={R:2}&amp;nav={R:4}"/>
                </rule>
            </rules>
        </rewrite>
    </system.webServer>
</configuration>
```

target.xml

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <rewrite>
            <rules>
                <rule name="Redirect to https" stopProcessing="true">
                    <match url="(.*)" />
                    <conditions>
                        <add input="{HTTPS}" pattern="off" ignoreCase="true" />
                    </conditions>
                    <action type="Redirect" url="https://{HTTP_HOST}{REQUEST_URI}" redirectType="Permanent" />
                </rule>

                <rule name="foo" stopProcessing="true">
                  <conditions logicalGrouping="MatchAll">
                    <add input="{HTTP_HOST}" pattern="foo.example.com" negate="false" />
                  </conditions>
                  <action type="Redirect" url="redacted" appendQueryString="false" redirectType="Permanent" />
                </rule>

            </rules>
        </rewrite>
    </system.webServer>
</configuration>
```

**Comment by CameronWills** [comment]
Good catch, thanks for logging the bug. I'll take a look at this bug in the next few days - I feel like this might be a tricky bug to fix.

And thanks for the feedback - I'm glad other devs are finding the tool useful.

**Comment by CameronWills** [comment]
This should be fixed now in version 0.2.8:
https://github.com/CameronWills/FatAntelope/releases/tag/0.2.8

I've tried to address some other less common cases as well.. I'll close this issue now, but reopen if you're still experiencing problems.
